libs/external-db-dynamodb/src/dynamo_data_provider.ts

```diff
@@ -25,45 +31,43 @@ class DataProvider {
 
         const { Items } = await this.query(dynamoRequests.findCommand(collectionName, filterExprWithProjection, limit), queryable)
 
-        return Items.map(patchFixDates)
+        return Items ? Items.map(patchFixDates) : []
     }
     
-    async count(collectionName, filter) {
+    async count(collectionName: string, filter: Filter): Promise<number> {
         const { filterExpr, queryable } = this.filterParser.transform(filter)
         const { Count } = await this.query(dynamoRequests.countCommand(collectionName, filterExpr), queryable)
-        return Count
+        return Count || 0
     }
 
-    async insert(collectionName, items) {
+    async insert(collectionName: string, items: Item[]): Promise<number> {
         validateTable(collectionName)
```

Comment:
why `valideTable` getting the `collectionName` as `any`? when you always pass the `colloectionName` as `string`?